controllers/application/service_monitor.go

```diff
@@ -32,17 +31,6 @@ func (r *ApplicationReconciler) reconcileServiceMonitor(ctx context.Context, app
 		return util.RequeueWithError(err)
 	}
 
-	if application.Spec.Prometheus == nil {
```

Comment:
What happens if you toggle istio on and off? Should this be changed to `!r.IsIstioEnabledForNamespace(ctx, application.Namespace)` instead of being removed so that it cleans up properly?

Reply:
I was considering the possibility of disabling istio in a Skiperator-managed application namespace rather slim, but I could absolutely re-add this block in order to lighten the load on grafana-agent when there's nothing to scrape.